lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -695,4 +746,41 @@ private Function<Integer, Object> getConverter(PrimitiveType primitiveType)
             super(columnPath, Integer.class);
         }
     }
+
+    /**
+     * Check if the predicateValue might be in the bloomfilter
+     *
+     * @param bloomFilter parquet bloomfilter.
+     * @param predicateValue effective discrete predicate value.
+     * @param sqlType Type that contains information about the type schema from connector's metadata
+     * @return true if the predicateValue might be in the bloomfilter, false if the predicateValue absolutely is not in the bloomfilter
+     */
+    @VisibleForTesting
+    public static boolean checkInBloomFilter(BloomFilter bloomFilter, Object predicateValue, Type sqlType)
+    {
+        // todo: support TIMESTAMP, and DECIMAL
+        if (sqlType == TINYINT || sqlType == SMALLINT || sqlType == INTEGER || sqlType == BIGINT || sqlType == DATE) {
+            return bloomFilter.findHash(bloomFilter.hash((int) asLong(predicateValue)));
+        }
+        else if (sqlType == DOUBLE) {
+            return bloomFilter.findHash(bloomFilter.hash(predicateValue));
```

Comment:
This seems redundant. since it called the interface:  

 long hash(Object value), so whether cast or not will not has impact. 

The implementation internally will do:

else if (value instanceof Double) {
      cacheBuffer.putDouble((Double) value)

Do you want me to do something like:

bloomFilter.hash(((Double) predicateValue).doubleValue()) ?  
Then it will call:

public long hash(double value) {
  cacheBuffer.putDouble(value);
  return doHash();
}

Reply:
yes, let's use `bloomFilter.hash(((Double) predicateValue).doubleValue())` here

Reply:
ok//

for learning purpose, can you briefly explain why we prefer hash(((Double) predicateValue).doubleValue()) over hash(predicateValue)?